### PR TITLE
feat: add ResourceReport dropdown button in TableDetail

### PR DESCRIPTION
**Pull request by dechoma**
### Summary of Changes

add button with dropdown list of linked table reports

dropdown_low

### Tests

N/A

### Documentation

N/A

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [x] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [x] PR includes a summary of changes, including screenshots of any UI changes. 
- [x] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public python functions and the classes in the PR contain docstrings that explain what it does
- [x] PR passes all tests documented in the [developer guide](https://github.com/lyft/amundsenfrontendlibrary/blob/master/docs/developer_guide.md#testing)

Resolve lyft/amundsen#525

### CODE REVIEW

**Comment by dechoma** [comment]
the PR should be well reviewed, surely need to improve the CSS styles, I also used button to handle drop-down list I don't know if this is a good idea

**Comment by jornh** [comment]
@dechoma 👏 great stuff!

Looks like there are a few errors, like https://travis-ci.com/github/lyft/amundsenfrontendlibrary/jobs/353457956#L742-L744 you need to fix to make the "Prettier" tool happy.

**Comment by ttannis** [comment]
@dechoma can you add a screen shot/gif if the interaction. 

Next steps here will be:
1. First reviewing the implementation in other repos, to ensure that no changes are requested that would affect this work
2. We'll have a designer weigh in on the UI
3. Request UI changes, if any, before final review.

**Comment by dechoma** [comment]
@jornh ,
Thanks for pointing that out, fixed in https://github.com/lyft/amundsenfrontendlibrary/pull/495/commits/d45455dd3d81940946719f340c8a43573829a6e1

**Comment by ttannis** [comment]
@dechoma, we've shared your PR with our designers and are currently waiting for feedback

File: amundsen_application/static/js/components/TableDetail/index.tsx [hunk 1]
```diff
@@ -204,6 +205,7 @@ export class TableDetail extends React.Component<
               <SourceLink tableSource={data.source} />
             </div>
             <div className="header-section header-buttons">
+              <TableReportsDropdown resourceReports={data.resource_reports} />
```
Comment (ttannis) [inline amundsen_application/static/js/components/TableDetail/index.tsx:208]: nit: Can you add a test in the `render` test here to verify `TableReportsDropdown` is rendered and additionally could check if it is given the correct properties https://github.com/lyft/amundsenfrontendlibrary/blob/master/amundsen_application/static/js/components/TableDetail/index.spec.tsx#L66

**Comment by ttannis** [comment]
@dechoma we haven't gotten design input yet, so I'll move this along from my perspective. Can you remove the `noCaret` option from the dropdown and ensure that there is  downward facing carat (if that isn't the default).

In case your interested, here's more documentation about that 3rd party dropdown component and other features:
https://react-bootstrap.github.io/components/dropdowns/

**Comment by dechoma** [comment]
@ttannis , on main issue @mgorsk1 initiated a discussion on whether it is better to replace the button with a tab in the table view, let's see how it evolves https://github.com/lyft/amundsen/issues/525#issuecomment-656481401

**Comment by ttannis** [comment]
> @ttannis , on main issue @mgorsk1 initiated a discussion on whether it is better to replace the button with a tab in the table view, let's see how it evolves [lyft/amundsen#525 (comment)](https://github.com/lyft/amundsen/issues/525#issuecomment-656481401)

Replied to the comment. But in summary the dropdown is preferred for now.
https://github.com/lyft/amundsen/issues/525#issuecomment-656884020

**Comment by ttannis** [comment]
> @dechoma we haven't gotten design input yet, so I'll move this along from my perspective. Can you remove the `noCaret` option from the dropdown and ensure that there is downward facing carat (if that isn't the default).
> 
> In case your interested, here's more documentation about that 3rd party dropdown component and other features:
> https://react-bootstrap.github.io/components/dropdowns/
> 
> edit: Also remove the `pullRight` option.

cc @mgorsk1 as well. We can re-review once these UI updates are made. Thanks!

**Comment by dechoma** [comment]
@ttannis thanks for remarks, I fixed mentioned options -  [caret also looks good](https://imagizer.imageshack.com/img922/4679/pKhB4G.png)

**Comment by ttannis** [comment]
> @ttannis thanks for remarks, I fixed mentioned options - [caret also looks good](https://imagizer.imageshack.com/img922/4679/pKhB4G.png)

Great, can you confirm the new UI with a GIF. And see previous comment on `TableDetail/index.tsx` about a unit test.

**Comment by dechoma** [comment]
@ttannis here's gif https://imagizer.imageshack.com/img922/3503/JYLmDv.gif, I also added the unit tests

**Pull request merged**